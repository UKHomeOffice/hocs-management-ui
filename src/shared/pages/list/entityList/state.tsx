import EntityListItem from '../../../models/entityListItem';

export interface State {
    entities: EntityListItem[];
    entitiesLoaded: boolean;
}
