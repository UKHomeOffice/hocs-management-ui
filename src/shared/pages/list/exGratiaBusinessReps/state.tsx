import EntityListItem from '../../../models/entityListItem';

export interface State {
    representatives: EntityListItem[];
    representativesLoaded: boolean;
}
