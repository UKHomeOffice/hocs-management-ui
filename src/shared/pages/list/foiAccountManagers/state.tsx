import EntityListItem from '../../../models/entityListItem';

export interface State {
    accountManagers: EntityListItem[];
    accountManagersLoaded: boolean;
}
