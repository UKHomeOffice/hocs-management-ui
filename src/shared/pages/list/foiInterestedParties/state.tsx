import EntityListItem from '../../../models/entityListItem';

export interface State {
    parties: EntityListItem[];
    partiesLoaded: boolean;
}
