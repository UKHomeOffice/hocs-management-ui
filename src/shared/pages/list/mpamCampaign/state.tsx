import EntityListItem from '../../../models/entityListItem';

export interface State {
    campaigns: EntityListItem[];
    campaignsLoaded: boolean;
}
