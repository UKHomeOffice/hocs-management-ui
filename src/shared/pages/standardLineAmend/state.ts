export default interface State {
    expiryDate: string;
    oldExpiryDate: string;
    displayName: string;
    topic: string;
}
