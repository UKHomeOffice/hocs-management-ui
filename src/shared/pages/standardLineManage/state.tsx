import StandardLineModel from '../../models/standardLineModel';

export interface State {
    activeStandardLines: StandardLineModel[];
    allStandardLines: StandardLineModel[];
    standardLinesLoaded: boolean;
    filter: string;
    excludeExpired: boolean;
}
