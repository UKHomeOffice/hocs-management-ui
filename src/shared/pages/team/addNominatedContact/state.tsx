import Item from '../../../models/item';
export interface State {
    inputValue: string;
    selectedContacts: Item[];
    teamName?: string;
    teamUUID: string;
}
