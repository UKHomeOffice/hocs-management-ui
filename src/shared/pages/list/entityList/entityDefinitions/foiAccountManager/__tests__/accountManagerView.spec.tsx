import React from 'react';
import { match, MemoryRouter } from 'react-router-dom';
import { createBrowserHistory, History, Location } from 'history';
import { act, waitFor, render, RenderResult } from '@testing-library/react';
import EntityListView from '../../../entityListView';
import * as ListService from '../../../../../../services/entityListService';
import { State } from '../../../state';
import * as useError from '../../../../../../hooks/useError';
import foiAccountManager from '../foiAccountManager';

let match: match<any>;
let history: History<any>;
let location: Location;
let mockState: State;

const FoiAccountManager = EntityListView(foiAccountManager);

jest.mock('../../../../../../services/entityListService', () => ({
    __esModule: true,
    getListItems: jest.fn().mockReturnValue(Promise.resolve({
        data: [{
            simpleName: 'testSimpleName1',
            uuid: 'testId1',
            title: 'testTitle1'
        }, {
            simpleName: 'testSimpleName2',
            uuid: 'testId2',
            title: 'testTitle2'
        }]
    }))
}));

const getListItemsSpy = jest.spyOn(ListService, 'getListItems');
const useReducerSpy = jest.spyOn(React, 'useReducer');
const useErrorSpy = jest.spyOn(useError, 'default');
const setMessageSpy = jest.fn();
const clearErrorsSpy = jest.fn();

const renderComponent = () => render(
    <MemoryRouter>
        <FoiAccountManager history={history} location={location} match={match}></FoiAccountManager>
    </MemoryRouter>
);

beforeEach(() => {
    history = createBrowserHistory();
    match = {
        isExact: true,
        params: {},
        path: '',
        url: ''
    };

    location = {
        hash: '',
        key: '',
        pathname: '',
        search: '',
        state: {}
    };
    const entities = [{
        simpleName: 'testSimpleName1',
        uuid: 'testId1',
        title: 'testTitle1',
        active: true
    }, {
        simpleName: 'testSimpleName2',
        uuid: 'testId2',
        title: 'testTitle2',
        active: true
    }];
    mockState = {
        entitiesLoaded: true,
        entities,
        showInactive: true,
        inactiveCount: 0,
        entitiesToDisplay: entities,
    };
    useReducerSpy.mockImplementationOnce(() => [mockState, jest.fn()]);
    useErrorSpy.mockImplementation(() => [{}, jest.fn(), clearErrorsSpy, setMessageSpy]);
    clearErrorsSpy.mockReset();
    setMessageSpy.mockReset();
});

describe('when the foiAccountManager EntityListView component is mounted', () => {
    it('should render with default props', async () => {
        expect.assertions(2);
        let wrapper: RenderResult;
        act(() => {
            wrapper = renderComponent();
        });

        await waitFor(() => {
            expect(getListItemsSpy).toHaveBeenCalled();
            expect(wrapper.container).toMatchSnapshot();
        });
    });
});
