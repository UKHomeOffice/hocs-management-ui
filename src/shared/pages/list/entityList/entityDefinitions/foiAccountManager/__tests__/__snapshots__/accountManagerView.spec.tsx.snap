// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the foiAccountManager EntityListView component is mounted should render with default props 1`] = `
<div>
  <div
    class="govuk-form-group"
  >
    <a
      class="govuk-back-link"
      href="/"
    >
      Back
    </a>
    <div>
      <h1
        class="govuk-heading-xl"
      >
        View and edit account managers
      </h1>
      <div>
        <table
          class="govuk-table"
        >
          <thead
            class="govuk-table__head"
          >
            <tr
              class="govuk-table__row"
            >
              <th
                class="govuk-table__header"
                scope="col"
              >
                Account manager name
              </th>
              <th
                class="govuk-table__header"
                scope="col"
              >
                Account manager code
              </th>
              <th
                class="govuk-table__header"
                scope="col"
              >
                Action
              </th>
            </tr>
          </thead>
          <tbody
            class="govuk-table__body"
          >
            <tr
              class="govuk-table__row"
            >
              <td
                class="govuk-table__cell"
              >
                testTitle1
              </td>
              <td
                class="govuk-table__cell"
              >
                testSimpleName1
              </td>
              <td
                class="govuk-table__cell"
              >
                <a
                  href="#"
                >
                  Amend
                </a>
              </td>
            </tr>
            <tr
              class="govuk-table__row"
            >
              <td
                class="govuk-table__cell"
              >
                testTitle2
              </td>
              <td
                class="govuk-table__cell"
              >
                testSimpleName2
              </td>
              <td
                class="govuk-table__cell"
              >
                <a
                  href="#"
                >
                  Amend
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <button
        class="govuk-button govuk-!-margin-right-1 add-team-members-button"
        data-module="govuk-button"
        type="submit"
      >
        Add new account manager
      </button>
    </div>
  </div>
</div>
`;
